src/upload/form/mod.rs

```diff
@@ -127,7 +128,17 @@ pub fn build(
                         plain_texts,
                     )?;
 
-                    modules_worker::build_form(&assets, session_config)
+                    if let Some(asset_manifest) = &asset_manifest {
```

Comment:
I'd do `let manifest_blob = asset_manifest.map(|m| get_asset_manifest_blob(&m))?` to get an `Option<String>` that you could pass into `build_form`

Might need to rearrange things to handle the `Result<String>` from `get_asset_manifest_blob`, or just use `.expect()` inside of  `get_asset_manifest_blob` so it can return a `String` (@nilslice, do y'all have a different convention here nowadays?) as there's no reason serializing the asset manifest into JSON should fail due to a user caused action.